website/draft_site/drafts/views.py

```diff
@@ -12,13 +14,52 @@
 from mtg_draft_ai.api import DraftInfo, Drafter, read_cube_toml
 from mtg_draft_ai.brains import GreedySynergyPicker
 
+import requests
+import toml
+
 
 CUBE_FILE = os.path.join(settings.DRAFTS_APP_DIR, 'cube_81183_tag_data.toml')
 CUBE_LIST = read_cube_toml(CUBE_FILE)
 CARDS_BY_NAME = {c.name: c for c in CUBE_LIST}
 PICKER_FACTORY = GreedySynergyPicker.factory(CUBE_LIST)
 LOGGER = logging.getLogger(__name__)
 
+IMAGE_URLS_FILE = os.path.join(settings.DRAFTS_APP_DIR, 'cube_81183_image_urls.toml')
+
+
+def _initialize_image_url_cache():
+    # Load cache from disk
+    cache = toml.load(IMAGE_URLS_FILE)
+
+    # Filter out cards which are no longer in the cube list
+    cache = {k: v for k, v in cache.items() if k in CARDS_BY_NAME}
+
+    # Get URLs for new cards
+    missing_cards = [name for name in CARDS_BY_NAME if name not in cache]
+    with ThreadPool(10) as tp:
+        urls = tp.map(_get_image_url, missing_cards)
+
+    for card_name, url in zip(missing_cards, urls):
+        cache[card_name] = url
+
+    # Write cache back out to disk
+    with open(IMAGE_URLS_FILE, 'w') as f:
+        toml.dump(cache, f)
+
+    return cache
+
+
+def _get_image_url(name):
+    r = requests.get('https://api.scryfall.com/cards/named', params={'exact': name})
+    card_json = r.json()
+    if 'card_faces' in card_json:
+        # Case for double-faced cards
+        return card_json['card_faces'][0]['image_uris']['normal']
```

Comment:
Something that might be neat to have in the future would be to have a way to show both sides of double faced cards.